src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -306,53 +307,112 @@ public static RestHighLevelClient getClientForElasticsearchCloud(@NonNull String
         return addressByParticipant;
     }
 
-    public static void writeWorkflow(@NonNull DDPInstance instance, @NonNull String ddpParticipantId, @NonNull String workflow, @NonNull String status) {
+    public static void writeWorkflow(@NonNull WorkflowForES workflowForES) {
+        String ddpParticipantId = workflowForES.getDdpParticipantId();
+        DDPInstance instance = workflowForES.getInstance();
         String index = instance.getParticipantIndexES();
-        if (StringUtils.isNotBlank(index)) {
-            try {
-                Map<String, Object> workflowMapES = getObjectsMap(index, ddpParticipantId, "workflows");
-                if (workflowMapES != null && !workflowMapES.isEmpty()) {
-                    List<Map<String, Object>> workflowListES = (List<Map<String, Object>>) workflowMapES.get("workflows");
-                    if (workflowListES != null && !workflowListES.isEmpty()) {
-                        boolean updated = false;
-                        for (Map<String, Object> workflowES : workflowListES) {
-                            if (workflow.equals(workflowES.get("workflow"))) {
-                                //update value in existing workflow
-                                workflowES.put("status", status);
-                                workflowES.put("date", SystemUtil.getISO8601DateString());
-                                updated = true;
-                                break;
-                            }
-                        }
-                        if (!updated) {
-                            //add workflow
-                            Map<String, Object> newWorkflowMap = new HashMap<>();
-                            newWorkflowMap.put("workflow", workflow);
-                            newWorkflowMap.put("status", status);
-                            newWorkflowMap.put("date", SystemUtil.getISO8601DateString());
-                            workflowListES.add(newWorkflowMap);
-                        }
+        if (StringUtils.isBlank(index)) {
+            return;
+        }
+        try {
+            Map<String, Object> workflowMapES = getObjectsMap(index, ddpParticipantId, ESObjectConstants.WORKFLOWS);
+            String workflow = workflowForES.getWorkflow();
+            String status = workflowForES.getStatus();
+            if (workflowMapES != null && !workflowMapES.isEmpty()) {
+                List<Map<String, Object>> workflowListES = (List<Map<String, Object>>) workflowMapES.get(ESObjectConstants.WORKFLOWS);
+                if (workflowListES != null && !workflowListES.isEmpty()) {
+                    boolean updated = false;
+                    if (workflowForES.getStudySpecificData() != null) {
+                        updateWorkflowStudySpecific(workflow, status, workflowListES, updated, workflowForES.getStudySpecificData());
+                    } else {
+                        updateWorkflow(workflow, status, workflowListES, updated);
```

Comment:
what's the point of passing `updated` primitive boolean here? changing its value in method `updateWorkflowStudySpecific` won't have any effect on `updated` on line 324